table/scanner.go

```diff
@@ -393,3 +398,62 @@ type FileScanTask struct {
 	DeleteFiles   []iceberg.DataFile
 	Start, Length int64
 }
+
+// ToArrowRecords returns the arrow schema of the expected records and an interator
+// that can be used with a range expression to read the records as they are available.
+// If an error is encountered, during the planning and setup then this will return the
+// error directly. If the error occurs while iterating the records, it will be returned
+// by the iterator.
+//
+// The purpose for returning the schema up front is to handle the case where there are no
+// rows returned. The resulting Arrow Schema of the projection will still be known.
+func (s *Scan) ToArrowRecords(ctx context.Context) (*arrow.Schema, iter.Seq2[arrow.Record, error], error) {
```

Comment:
How about slightly longer variable names? :)
```suggestion
func (scan *Scan) ToArrowRecords(ctx context.Context) (*arrow.Schema, iter.Seq2[arrow.Record, error], error) {
```